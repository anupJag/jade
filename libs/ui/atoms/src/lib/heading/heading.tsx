import React, { FC } from 'react';
import * as S from './heading.styled';

export type ValidHeadings = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export type HeadingProps = {
  tagName: ValidHeadings;
};
export const Heading: FC<HeadingProps> = ({ tagName, children }) => {
  const Tag = S[tagName];
  return <Tag>{children}</Tag>;
};
