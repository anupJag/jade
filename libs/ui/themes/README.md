# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
