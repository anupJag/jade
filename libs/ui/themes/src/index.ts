export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
