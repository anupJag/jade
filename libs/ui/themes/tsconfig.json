{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
