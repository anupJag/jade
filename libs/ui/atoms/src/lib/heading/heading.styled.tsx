import { styled } from '@jade/ui/themes';

export const h1 = styled.h1`
  ${props => props.theme.themeFonts.heading}
  font-size: ${props => props.theme.themFontSizes.h1};
  font-weight: ${props => props.theme.themeFontWeights.bold}
`;
export const h2 = styled.h2`
  ${props => props.theme.themeFonts.heading}
  font-size: ${props => props.theme.themFontSizes.h2};
  font-weight: ${props => props.theme.themeFontWeights.light}
`;
export const h3 = styled.h3`
  ${props => props.theme.themeFonts.heading}
  font-size: ${props => props.theme.themFontSizes.h3};
  font-weight: ${props => props.theme.themeFontWeights.light}
`;
export const h4 = styled.h4`
  ${props => props.theme.themeFonts.heading}
  font-size: ${props => props.theme.themFontSizes.h4};
  font-weight: ${props => props.theme.themeFontWeights.light}
`;
export const h5 = styled.h5`
  ${props => props.theme.themeFonts.heading}
  font-size: ${props => props.theme.themFontSizes.h5};
  font-weight: ${props => props.theme.themeFontWeights.light}
`;
export const h6 = styled.h6`
  ${props => props.theme.themeFonts.heading}
  font-size: ${props => props.theme.themFontSizes.h6};
  font-weight: ${props => props.theme.themeFontWeights.light}
`;
