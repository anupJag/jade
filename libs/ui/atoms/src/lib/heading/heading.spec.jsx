import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider } from '@jade/ui/themes';
import { Heading } from './heading';

describe('<Styled Component : Heading />', () => {
  let HeadingComponent;
  const listOfStyledTags = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];

  beforeEach(() => {
    HeadingComponent = render(
      <ThemeProvider>
        {listOfStyledTags.map(tag => (
          <Heading key={tag} tagName={tag}>
            {tag}
          </Heading>
        ))}
      </ThemeProvider>,
    );
  });

  /**
   * Capturing Sanpshot : HeadingComponent
   *
   *  To check style / structure changes in Styled Components
   */
  test("snapshot check for HeadingComponent's Styled components", () => {
    const { asFragment } = HeadingComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
