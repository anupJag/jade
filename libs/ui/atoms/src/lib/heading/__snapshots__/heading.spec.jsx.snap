// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Styled Component : Heading /> snapshot check for HeadingComponent's Styled components 1`] = `
<DocumentFragment>
  <h1
    class="css-7imb5c"
  >
    h1
  </h1>
  <h2
    class="css-zhbmh3"
  >
    h2
  </h2>
  <h3
    class="css-v7zg6k"
  >
    h3
  </h3>
  <h4
    class="css-mcy282"
  >
    h4
  </h4>
  <h5
    class="css-1nhnene"
  >
    h5
  </h5>
  <h6
    class="css-1nhnene"
  >
    h6
  </h6>
</DocumentFragment>
`;
